app/services/order_detail_batch_updater.rb

```diff
@@ -121,6 +124,9 @@ def update_order_details_from_params
   def update_all(attribute, value)
     order_details.each do |order_detail|
       order_detail.public_send("#{attribute}=", value)
+      if order_detail.assigned_user_id_changed? && order_detail.assigned_user_id.present?
```

Comment:
We could conceivably pull this up a level or two and do something like`order_details.select { |od| od.assigned_user_id_changed? && od.assigned_user_id.present? }`. Maybe to the `update!` method right before the `all?(&:save!)`? This seems a little buried to me.